packages/rum-recorder/test/utils.ts

```diff
@@ -117,3 +119,283 @@ export function collectAsyncCalls<F extends jasmine.Func>(spy: jasmine.Spy<F>) {
+  idAttribute?: string
+  // Select the first node with the given text content from the initial full snapshot
+  text?: string
+}
+
+interface ExpectedTextMutation {
+  // Reference to the node where the mutation happens
+  node: ExpectedNode
+  // New text value
+  value: string
+}
+
+interface ExpectedAttributeMutation {
+  // Reference to the node where the mutation happens
+  node: ExpectedNode
+  // Updated attributes
+  attributes: {
+    [key: string]: string | null
+  }
+}
+
+interface ExpectedRemoveMutation {
+  // Reference to the removed node
+  node: ExpectedNode
+  // Reference to the parent of the removed node
+  parent: ExpectedNode
+}
+
+interface ExpectedAddMutation {
+  // Partially check for the added node properties.
+  node: ExpectedNode
+  // Reference to the parent of the added node
+  parent: ExpectedNode
+  // Reference to the sibling of the added node
+  next?: ExpectedNode
+}
+
+interface ExpectedMutationsPayload {
+  texts?: ExpectedTextMutation[]
+  attributes?: ExpectedAttributeMutation[]
+  removes?: ExpectedRemoveMutation[]
+  adds?: ExpectedAddMutation[]
+}
+
+/**
+ * ExpectedNode is a helper class to build a serialized Node tree to be used to validate mutations.
+ * For now, its purpose is limited to specifying child nodes.
+ */
+class ExpectedNode {
+  constructor(private node: Omit<SerializedNodeWithId, 'childNodes'> & { childNodes?: ExpectedNode[] }) {}
+
+  withChildren(...childNodes: ExpectedNode[]): ExpectedNode {
+    return new ExpectedNode({ ...this.node, childNodes })
+  }
+
+  getId() {
+    return this.node.id
+  }
+
+  toSerializedNodeWithId() {
+    const { childNodes, ...result } = this.node
+    if (childNodes) {
+      ;(result as any).childNodes = childNodes.map((node) => node.toSerializedNodeWithId())
+    }
+    return result as SerializedNodeWithId
+  }
+}
+
+type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>
+
+/**
+ * Based on an serialized initial document, it returns:
+ *
+ * * a set of utilities functions to select existing nodes (selectNode) or create new nodes
+ * (newNode) to help build the expected mutations object
+ *
+ * * a 'validate' function to actually validate a mutation payload against an expected mutation
+ * object.
+ */
+export function createMutationPayloadValidator(initialDocument: SerializedNodeWithId) {
+  let maxNodeId = findMaxNodeId(initialDocument)
+
+  /**
+   * Creates a new node based on input parameter, with sensible default properties, and an
+   * automatically computed 'id' attribute based on the previously created nodes.
+   */
+  function newNode(node: Optional<ElementNode, 'childNodes' | 'attributes'>): ExpectedNode
+  function newNode(node: TextNode): ExpectedNode
+  function newNode(node: Partial<SerializedNode>) {
+    maxNodeId += 1
+    if (node.type === NodeType.Element) {
+      node.attributes ||= {}
+      node.childNodes = []
+    }
+    return new ExpectedNode({
+      ...node,
+      id: maxNodeId,
+    } as any)
+  }
```

Comment:
FMU, the idea is to create an ExpectedNode corresponding to a node added during the test.
Could it be clearer/more consistent with a naming like `addedNode`?

Reply:
I think `newNode` is clearer than `addedNode`. It represent a node that is not in the initial serialized document. We could have a node in the initial document that is removed and readded during the test, so `addedNode` might be confusing.

Reply:
OK, counter-proposal: 
- `selectNode` -> `expectInitialNode`
- `newNode` -> `expectNewNode`

Reply:
Alright! It is a bit more verbose but I'm fine with it